app/controllers/guarantee/LiabilityAmountController.scala

```diff
@@ -64,13 +65,16 @@ class LiabilityAmountController @Inject() (
     .andThen(getMandatoryPage(CurrencyPage(index)))
     .async {
       implicit request =>
+        lazy val redirect = controllers.guarantee.routes.AddDefaultLiabilityAmountController.onPageLoad(lrn, mode, index)
         form
           .bindFromRequest()
           .fold(
             formWithErrors => Future.successful(BadRequest(view(formWithErrors, lrn, mode, index, request.arg.symbol))),
-            value => {
-              implicit val navigator: UserAnswersNavigator = navigatorProvider(mode, index)
-              LiabilityAmountPage(index).writeToUserAnswers(value).updateTask().writeToSession().navigate()
+            {
+              case v: BigDecimal if v == 0 => Future.successful(Redirect(redirect))
+              case value =>
+                implicit val navigator: UserAnswersNavigator = navigatorProvider(mode, index)
```

Comment:
I think this logic can maybe we handled in the navigator. We can read the value LiabilityAmountPage(index) there and handle the navigation. You could also create a custom domain model for this particular scenario.